registry/handlers/health_test.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		t.Fatalf("error unmarshaling json: %v", err)
+	}
+	if len(decoded) != 1 {
+		t.Fatal("expected 1 item in returned json")
+	}
+	if decoded[tmpfile.Name()] != "file exists" {
+		t.Fatal(`did not get "file exists" result for health check`)
+	}
+
+	os.Remove(tmpfile.Name())
+
+	<-time.After(2 * interval)
+	resp, err = http.Get(debugServer.URL + "/debug/health")
+	if err != nil {
+		t.Fatalf("error performing HTTP GET: %v", err)
+	}
+	body, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("error reading HTTP body: %v", err)
+	}
+	resp.Body.Close()
+	var decoded2 map[string]string
+	err = json.Unmarshal(body, &decoded2)
+	if err != nil {
+		t.Fatalf("error unmarshaling json: %v", err)
+	}
+	if len(decoded2) != 0 {
+		t.Fatal("expected 0 items in returned json")
+	}
+}
+
+func TestHTTPHealthCheck(t *testing.T) {
+	// In case other tests registered checks before this one
+	health.UnregisterAll()
+
+	interval := time.Second
+	threshold := 3
+
+	stopFailing := make(chan struct{})
+
+	checkedServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "HEAD" {
+			t.Fatalf("expected HEAD request, got %s", r.Method)
+		}
+		select {
+		case <-stopFailing:
+			w.WriteHeader(http.StatusOK)
+		default:
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	}))
+
+	config := configuration.Configuration{
+		Storage: configuration.Storage{
+			"inmemory": configuration.Parameters{},
+		},
+		Health: configuration.Health{
+			HTTPCheckers: []configuration.HTTPChecker{
+				{
+					Interval:  interval,
+					URI:       checkedServer.URL,
+					Threshold: threshold,
+				},
+			},
+		},
+	}
+
+	ctx := context.Background()
+
+	app := NewApp(ctx, config)
+	app.RegisterHealthChecks()
+
+	debugServer := httptest.NewServer(nil)
+
+	for i := 0; ; i++ {
+		<-time.After(interval)
+
+		resp, err := http.Get(debugServer.URL + "/debug/health")
+		if err != nil {
+			t.Fatalf("error performing HTTP GET: %v", err)
+		}
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("error reading HTTP body: %v", err)
+		}
+		resp.Body.Close()
+		var decoded map[string]string
+		err = json.Unmarshal(body, &decoded)
+		if err != nil {
+			t.Fatalf("error unmarshaling json: %v", err)
+		}
+
+		if i < threshold-1 {
+			// definitely shouldn't have hit the threshold yet
+			if len(decoded) != 0 {
+				t.Fatal("expected 1 items in returned json")
+			}
+			continue
+		}
+		if i < threshold+1 {
+			// right on the threshold - don't expect a failure yet
+			continue
+		}
+
+		if len(decoded) != 1 {
+			t.Fatal("expected 1 item in returned json")
+		}
+		if decoded[checkedServer.URL] != "downstream service returned unexpected status: 500" {
+			t.Fatal("did not get expected result for health check")
+		}
+
+		break
+	}
+
+	// Signal HTTP handler to start returning 200
+	close(stopFailing)
+
+	<-time.After(2 * interval)
+	resp, err := http.Get(debugServer.URL + "/debug/health")
```

Comment:
My only issue is that these tests are harder to maintain over time and I suspect we will make changes to the http interface.

It is probably okay to leave this since it has already been done.